saleor/graphql/checkout/mutations/checkout_lines_update.py

```diff
@@ -221,3 +224,30 @@ def _get_variants_from_lines_input(cls, lines: list[dict]) -> list[ProductVarian
     @classmethod
     def _get_grouped_lines_data(cls, lines, existing_lines_info):
         return group_lines_input_data_on_update(lines, existing_lines_info)
+
+    @classmethod
+    def _validate_gift_line(
+        cls, lines_data: list[CheckoutLineData], lines_info: list[CheckoutLineInfo]
+    ):
+        gift_line_ids = [
+            str(line_info.line.id) for line_info in lines_info if line_info.line.is_gift
+        ]
+        gift_lines_with_quantity_to_update = [
+            line
+            for line in lines_data
+            if line.line_id in gift_line_ids and line.quantity_to_update
```

Comment:
In line 233 you, are making `str(line_info.line.id)`, here you're checking `line.line_id in gift_line_ids`. This could mean, incorrect types, right? Like id in uuid vs str(uuid)

Reply:
`CheckoutLineInfo.line.id` is type of `UUID`, while `CheckoutLineData.line_id` is type of `str`